internal/datacoord/index_service.go

```diff
@@ -162,18 +166,157 @@ func (s *Server) createIndexForSegmentLoop(ctx context.Context) {
 	}
 }
 
-func (s *Server) getFieldNameByID(ctx context.Context, collID, fieldID int64) (string, error) {
+func (s *Server) getFieldNameByID(schema *schemapb.CollectionSchema, collID, fieldID int64) (string, error) {
+	for _, field := range schema.GetFields() {
+		if field.FieldID == fieldID {
+			return field.Name, nil
+		}
+	}
+	return "", nil
+}
+
+func (s *Server) getSchema(ctx context.Context, collID int64) (*schemapb.CollectionSchema, error) {
 	resp, err := s.broker.DescribeCollectionInternal(ctx, collID)
 	if err != nil {
-		return "", err
+		return nil, err
 	}
+	return resp.GetSchema(), nil
+}
 
-	for _, field := range resp.GetSchema().GetFields() {
-		if field.FieldID == fieldID {
-			return field.Name, nil
+func (s *Server) isJsonField(schema *schemapb.CollectionSchema, collID, fieldID int64) (bool, error) {
+	for _, f := range schema.Fields {
+		if f.FieldID == fieldID {
+			return typeutil.IsJSONType(f.DataType), nil
 		}
 	}
-	return "", nil
+	return false, nil
+}
+
+const JSON_PATH_KEY = "json_path"
+const JSON_CAST_KEY = "json_cast_type"
+
+func getIndexParam(indexParams []*commonpb.KeyValuePair, key string) (string, error) {
+	for _, p := range indexParams {
+		if p.Key == JSON_PATH_KEY {
+			return p.Value, nil
+		}
+	}
+	return "", merr.WrapErrParameterInvalidMsg("%s not found", key)
+}
+
+func setIndexParam(indexParams []*commonpb.KeyValuePair, key, value string) {
+	for _, p := range indexParams {
+		if p.Key == key {
+			p.Value = value
+		}
+	}
+}
+
+func (s *Server) checkJsonParams(req *indexpb.CreateIndexRequest) error {
+	var jsonPath = ""
+	var jsonCastType = schemapb.DataType_None
+	for _, p := range req.IndexParams {
+		if p.Key == JSON_PATH_KEY {
+			jsonPath = p.Value
+		} else if p.Key == JSON_CAST_KEY {
+			v, err := strconv.ParseInt(p.Value, 10, 64)
+			if err != nil {
+				return err
+			}
+			jsonCastType = schemapb.DataType(v)
+		}
+	}
+
+	if jsonPath == "" {
+		return merr.WrapErrParameterInvalidMsg("%s is empty", JSON_PATH_KEY)
+	}
+	if jsonCastType == schemapb.DataType_None {
+		return merr.WrapErrParameterInvalidMsg("%s is empty", JSON_CAST_KEY)
+	}
+
+	// TODO: sunby, check json path
+	// check json cast type
+	switch jsonCastType {
+	case schemapb.DataType_Bool, schemapb.DataType_Int8, schemapb.DataType_Int16, schemapb.DataType_Int32, schemapb.DataType_Int64, schemapb.DataType_Float, schemapb.DataType_Double,
+		schemapb.DataType_String, schemapb.DataType_VarChar:
+	default:
+		return merr.WrapErrParameterInvalidMsg("invalid json cast type %s", jsonCastType.String())
+	}
+
+	// check duplicate json path
+	exists := s.meta.indexMeta.GetFieldIndexes(req.GetCollectionID(), req.GetFieldID(), req.GetIndexName())
```

Comment:
Is it allowed to create duplicated indexes with the same json path and different index names?

Reply:
> Is it allowed to create duplicated indexes with the same json path and different index names?

no, it is checked in checkJsonParams